src/geotiff-optimizer/optimize-geotiff.js

```diff
@@ -0,0 +1,54 @@
+import { exec } from 'child_process';
+import fs from 'fs';
+import path from 'path';
+
+/**
+ * Executes a shell command and return it as a Promise.
+ * Kudos to https://ali-dev.medium.com/how-to-use-promise-with-exec-in-node-js-a39c4d7bbf77
+ *
+ * @param cmd {String} The command to execute
+ * @return {Promise<String>} A Promise returning the console output
+ */
+const execShellCommand = (cmd) => {
+    return new Promise((resolve, reject) => {
+        exec(cmd, (error, stdout, stderr) => {
+            if (error) {
+                console.warn(error);
+                reject(error);
+            }
+            resolve(stdout ? stdout : stderr);
+        });
+    });
+}
+
+/**
+ * Optimize a GeoTIFF for the cloud by converting it to a COG.
+ *
+ * Also computes raster overviews.
+ * Relies on GDAL >= 3.4
```

Comment:
the github action, test.yaml we install gdal via `apt install gdal-bin`.  Will it be > 3.4?

Reply:
good point. I looked it up: https://packages.ubuntu.com/search?suite=jammy&searchon=names&keywords=gdal
Indeed Ubuntu `22.04` will install GDAL `3.4.1`